tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/mdspan_to_view_mapper_kk.hpp

```diff
@@ -22,19 +30,72 @@ struct LayoutMapper<std::experimental::layout_right> {
 };
 
 template<class MDSpan>
-std::enable_if_t<MDSpan::rank()==1, Kokkos::View<typename MDSpan::element_type*>>
+std::enable_if_t<
+  MDSpan::rank()==1
+  && !is_complex_v<typename MDSpan::element_type>,
```

Comment:
This will fail for `mdspan<const std::complex<T>, ...>`.  If that's not the intent, then a potential fix would be to use `MDSpan::value_type` (which removes any `const` and `volatile` qualifications) instead of `MDSpan::element_type`.  You would need to use `value_type` instead of `element_type` everywhere you use `is_complex_v`.

Another option would be to define `is_complex_v` to use `remove_cv_t` on its argument.  I personally wouldn't mind that, but my view is that using `MDSpan::value_type` better expresses the desired intent, which is to say something about the value type (with any `const` or `volatile` qualifiers removed).

In my view, an even better approach would be not to specialize `mdspan_to_view` on whether the value type is complex.  Instead, you could write a function just to convert the input's raw pointer to a raw pointer that Kokkos understands, and then use that function everywhere.  Godbolt link with an example: https://godbolt.org/z/Pb39fq1so

```c++
template<class ValueType>
ValueType* toKokkosPointer(ValueType* p) { return p; }

template<class RealType>
auto toKokkosPointer(std::complex<RealType>* p) {
    assertAlignment(p); // whatever this should be
    return reinterpret_cast<Kokkos::complex<RealType>*>(p);
}
```

There are two reasons to do it this way.

1. It orthogonalizes different reasons to specialize into different layers.  One layer handles raw pointer conversion, and the other layer handles rank 1, rank 2, rank 3, etc.
2. It opens up an extension point for other kinds of pointer conversions that you might need in the future.

Reply:
nice suggestion, I followed it in this new revised impl !